File: src/main/java/com/github/wgzhao/dbquery/entities/QueryConfig.java
package com.github.wgzhao.dbquery.entities;


import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table
@Setter
@Getter
public class QueryConfig extends BaseEntity
{
    @Id
    private String selectId;

    private String querySql;

    private String dataSource;

    private int cacheTime;

    private String note;

    private boolean enabled;

    private boolean enableCache;

    public boolean isEnable() {
        return enabled;
    }
}

File: src/main/kotlin/com/github/wgzhao/dbquery/entities/QueryConfig.kt
package com.github.wgzhao.dbquery.entities

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.sql.Timestamp

@Entity
@Table(name = "query_config")
data class QueryConfig(
    @Id
    var selectId: String,
    var querySql: String,
    var dataSource: String,
    var cacheTime: Int = 0,
    var note: String? = null,
    var isEnable: Boolean = false,
    var enableCache: Boolean = false,
    val createdAt: Timestamp = Timestamp(System.currentTimeMillis()),
    val updatedAt: Timestamp = Timestamp(System.currentTimeMillis())
)
